### PR TITLE
Grafana, Telegraf, InfluxDB optional

### LINKED ISSUE #1591: Make grafana optional
## Describe the feature
Allow grafana/telegraf monitoring be enabled/disabled

```
# Options for the monitoring and alerting stack 
monitoring:
  azure_monitor_agent: false # Install Azure Monitor Agent on static infra VMs
  telegraf: false # Install telegraf on static infra VMs and dynamic compute nodes
  grafana: false # Deploy a Grafana instance with pre-defined dashboards

```

### PR DESCRIPTION
**Pull request by vgamayunov**
config.yml settings for the monitoring and alerting stack:
```yml
monitoring:
  azure_monitor_agent: false # Install Azure Monitor Agent on static infra VMs
  telegraf: true # Install telegraf on static infra VMs and dynamic compute nodes
  Grafana: true # Deploy a Grafana instance with pre-defined dashboards
```
`grafana` and `telegraf` are true by default
`azure_monitor_agent` is false by default
property `install_agent` has been renamed to `azure_monitor_agent`

close #1591

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -135,6 +135,21 @@ function use_local_users()
   fi
 }
 
+function use_grafana_telegraf()
+{
+  local use_grafana
+  local use_telegraf
+  use_grafana=$(yq eval '.monintoring.grafana' config.yml)
```
Comment (xpillons) [inline install.sh:142]: Typo here, should be monitoring

Comment (xpillons) [reply install.sh:142]: monitoring may not be present

File: config.schema.json [hunk 2]
```diff
@@ -261,10 +261,12 @@
...
             "additionalProperties": false,
             "properties": {
-                "install_agent": {"type": "boolean" }
+                "azure_monitor_agent": {"type": "boolean" },
+                "telegraf": {"type": "boolean", "default": true },
+                "grafana": {"type": "boolean", "default": true }
             },
             "required": [
-                "install_agent"
+                "azure_monitor_agent"
```
Comment (xpillons) [inline config.schema.json:269]: for me should be optional default to false

**Review by xpillons — CHANGES_REQUESTED** [review]

File: install.sh (hunk 1 above)
Comment (vgamayunov) [reply install.sh]: fixed the typo. 
when not present, this evaluates to 'null'. action is taken if the value is 'false'

File: install.sh [hunk 3]
```diff
@@ -139,7 +139,7 @@ function use_grafana_telegraf()
 {
   local use_grafana
   local use_telegraf
-  use_grafana=$(yq eval '.monintoring.grafana' config.yml)
+  use_grafana=$(yq eval '.monitoring.grafana' config.yml)
   use_telegraf=$(yq eval '.monintoring.telegraf' config.yml)
```
Comment (xpillons) [inline install.sh]: same typo for telegraf

**Review by xpillons — CHANGES_REQUESTED** [review]

**Review by xpillons — APPROVED** [review]

**Pull request merged**